runtime/gateway.go

```diff
@@ -276,19 +281,25 @@ func (gateway *Gateway) setupConfig(config *StaticConfig) {
 
 func (gateway *Gateway) setupMetrics(config *StaticConfig) error {
 	metricsType := config.MustGetString("metrics.type")
+	env := config.MustGetString("env")
+	hostname := getHostname()
+	service := config.MustGetString("metrics.tally.service")
 
 	var metricsBackend tally.CachedStatsReporter
 	if metricsType == "m3" {
 		if gateway.metricsBackend != nil {
 			panic("expected no metrics backend in gateway.")
 		}
 
-		env := config.MustGetString("env")
-
-		commonTags := map[string]string{"env": env}
+		// TODO: Why aren't common tags emitted?
+		// m3.Options.Env and m3.Options.Service adds 'env' and 'service' common tags.
+		// Providing 'host' common tag to prevent m3.Options.IncludeHost from returning an error on 'unknown'host.
+		commonTags := map[string]string{
+			"host": hostname,
```

Comment:
we do not want the host tag at all, please pass the 'env' tag explicitly, it's less confusing, we can drop the `Env` field in `m3.Options`.

It's also verify concerning that the `env` common tag does not show up in the health test with metrics.